drivers/display/display_rm68200.c

```diff
@@ -0,0 +1,1430 @@
+
+	param[0] = MCS_GAMMA_VN147;
+	param[1] = 0x15;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN175;
+	param[1] = 0x0B;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN203;
+	param[1] = 0x10;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN231;
+	param[1] = 0x07;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN239;
+	param[1] = 0x0F;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN247;
+	param[1] = 0x12;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN251;
+	param[1] = 0x0C;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MCS_GAMMA_VN255;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	/* Page 0 */
+	param[0] = DCS_CMD_PAGE;
+	param[1] = DCS_CMD_PAGE_UCS;
+	rm68200_generic_write(dev, param, 2);
+
+	param[0] = MIPI_DCS_EXIT_SLEEP_MODE;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	k_sleep(K_MSEC(200));
+
+	param[0] = MIPI_DCS_SET_DISPLAY_ON;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	k_sleep(K_MSEC(100));
+
+	param[0] = MIPI_DCS_WRITE_MEMORY_START;
+	rm68200_generic_write(dev, param, 1);
+
+	param[0] = MIPI_DCS_SET_TEAR_ON;
+	param[1] = 0x00;
+	rm68200_generic_write(dev, param, 2);
+
+	k_sleep(K_MSEC(200));
+
+	drv_data->bl_gpio = device_get_binding(config->bl_label);
+	if (drv_data->bl_gpio != NULL) {
+		ret = gpio_pin_configure(drv_data->bl_gpio,
+					 config->bl_pin,
+					 GPIO_OUTPUT_INACTIVE |
+					 config->bl_flags);
+		if (ret < 0) {
+			LOG_ERR("Could not configure bl GPIO (%d)", ret);
+			return ret;
+		}
+
+		/* turn the backlight on */
+		if (drv_data->bl_gpio) {
+			ret = gpio_pin_set(drv_data->bl_gpio,
+						 config->bl_pin,
+						 1);
+			if (ret) {
+				return ret;
+			}
+		}
+	}
+
+	return 0;
+}
+
+#define RM68200_PANEL(id)							\
+	static const struct rm68200_config rm68200_config_##id = {		\
+		.mipi_dsi_label = DT_INST_BUS_LABEL(id),			\
+		.reset_label = UTIL_AND(DT_INST_NODE_HAS_PROP(id, reset_gpios),	\
+					DT_INST_GPIO_LABEL(id, reset_gpios)),	\
+		.reset_pin = UTIL_AND(DT_INST_NODE_HAS_PROP(id, reset_gpios),	\
+				      DT_INST_GPIO_PIN(id, reset_gpios)),	\
+		.reset_flags = UTIL_AND(DT_INST_NODE_HAS_PROP(id, reset_gpios),	\
+					DT_INST_GPIO_FLAGS(id, reset_gpios)),	\
+		.pwr_label = UTIL_AND(DT_INST_NODE_HAS_PROP(id, pwr_gpios),	\
+				      DT_INST_GPIO_LABEL(id, pwr_gpios)),	\
+		.pwr_pin = UTIL_AND(DT_INST_NODE_HAS_PROP(id, pwr_gpios),	\
+				    DT_INST_GPIO_PIN(id, pwr_gpios)),		\
+		.pwr_flags = UTIL_AND(DT_INST_NODE_HAS_PROP(id, pwr_gpios),	\
+				      DT_INST_GPIO_FLAGS(id, pwr_gpios)),	\
+		.bl_label = UTIL_AND(DT_INST_NODE_HAS_PROP(id, bl_gpios),	\
+				     DT_INST_GPIO_LABEL(id, bl_gpios)),		\
+		.bl_pin = UTIL_AND(DT_INST_NODE_HAS_PROP(id, bl_gpios),		\
+				   DT_INST_GPIO_PIN(id, bl_gpios)),		\
+		.bl_flags = UTIL_AND(DT_INST_NODE_HAS_PROP(id, bl_gpios),	\
+				     DT_INST_GPIO_FLAGS(id, bl_gpios)),		\
```

Comment:
I was taking a look at the driver since I was already here for the bindings and noticed this. It should be using GPIO_DT_SPEC_GET_OR() in all cases.

Reply:
Thanks for your review, I have update the PR.